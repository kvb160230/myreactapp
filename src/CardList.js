import React from 'react';
import Card from './Card';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
    style: {
        display: "flex",
        flexDirection: "row",
        flexWrap: "wrap",
        padding: "25px",
        maxWidth: "100%",
        justifyContent: "center",
    },
  });
  

const CardList = (props) => {
    const classes = useStyles();
    const {people} = props;
    console.log(people);

    return (
        <div className={classes.style}>
        {
            people.length > 0 ? 
            people.map((user, i) => {
                return (
                    <Card 
                    key={i} 
                    id = {people[i].id} 
                    name = {people[i].name} 
                    nickname = {people[i].nickname}
                    title = {people[i].title}
                    email = {people[i].email}
                    team = {people[i].team}
                    default = {people[i].default}
                    />
                );
            }): "No employee with the matching name found"
        }
        </div>
    );
}

export default CardList;