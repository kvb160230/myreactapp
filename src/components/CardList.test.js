import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

it('expect to render CardList component', () => {

    const mockPeople = [
        {
            id: 1,
            name: 'Bruce Wayne',
            nickname: 'The Dark Knight‎‎‎',
            title: 'CEO',
            email: 'bruce@waynetech.com',
            team: 'Operations',
            default: 'test'
        }
    ]
  const cardComponent = shallow(<CardList people={mockPeople} />);
  expect(cardComponent.debug()).toMatchSnapshot();
});

