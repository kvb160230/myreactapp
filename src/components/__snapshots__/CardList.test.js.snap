// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
"<div className=\\"makeStyles-style-1\\">
  <MediaCard id={1} name=\\"Bruce Wayne\\" nickname=\\"The Dark Knight‎‎‎\\" title=\\"CEO\\" email=\\"bruce@waynetech.com\\" team=\\"Operations\\" default=\\"test\\" />
</div>"
`;
